hfst_optimized_lookup/_hfst_optimized_lookup.pyx

```diff
@@ -58,3 +64,19 @@ cdef class PyTransducerFile:
     def __dealloc__(self):
         del self.c_tf
 
+
+def _parse_analysis(letters_and_tags):
+    prefix_tags = []
+    lemma_chars = []
+    suffix_tags = []
+    affixes = prefix_tags
+
+    for symbol in letters_and_tags:
+        if len(symbol) == 1:
+            affixes = suffix_tags
```

Comment:
Is this making a reference or a copy? I'm skeptical that it does what I think it should be doing

Reply:
It's intentionally making a reference here. I guess this algorithm is overly clever. Basically, `affixes` is initially equivillent to `prefixes_tags`, and then, as soon as one lemma character is seen, it switches over to `suffix_tags`.

Is there some way I can make this algorithm more clear?

Reply:
The intentions are clear and if it works, I'm comfortable with it. It was unfamiliar so I wanted clarification before approving it.